be/src/io/fs/azure_obj_storage_client.cpp

```diff
@@ -251,4 +256,21 @@ ObjectStorageResponse AzureObjStorageClient::delete_objects_recursively(
     }
     return {};
 }
+
+std::string AzureObjStorageClient::generate_presigned_url(const ObjectStoragePathOptions& opts,
+                                                          int64_t expiration_secs,
+                                                          const S3ClientConf& conf) {
+    Azure::Storage::Sas::BlobSasBuilder sas_builder;
+    sas_builder.ExpiresOn =
+            std::chrono::system_clock::now() + std::chrono::seconds(expiration_secs);
+    sas_builder.BlobContainerName = opts.bucket;
+    sas_builder.BlobName = opts.key;
+    sas_builder.Resource = Azure::Storage::Sas::BlobSasResource::Blob;
+    sas_builder.SetPermissions(Azure::Storage::Sas::BlobSasPermissions::Write);
```

Comment:
We also need "Read" permission here.
Make sure that this signature is constrained by the specific blob.

And USE HTTPS only here
```
sas_biulder.Protocol = SasProtocol.HttpsOnly;
```